syntax = "proto3";

package data;

// Request message containing CSV data as bytes.
message RawData {
    bytes csv_content = 1;
}


// Response message containing cleaned data.
message CleanedData {
    repeated double x_train = 1;
    repeated double y_train = 2;
    repeated double x_test = 3;
    repeated double y_test = 4;
    repeated string dates_train = 5;
    repeated string dates_test = 6;
}


// The DataService definition.
service DataService {
    // Sends CSV content and receives cleaned data.
    rpc CleanData (RawData) returns (CleanedData);
}
